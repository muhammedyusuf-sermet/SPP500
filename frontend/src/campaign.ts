export interface ICampaignState {
	Id?: string,
	Name: string,
	Description: string;
	// Monsters: Monster.IMonsterState[];
}
