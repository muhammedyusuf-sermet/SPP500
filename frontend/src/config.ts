export const API_URL = "https://http://3.18.65.138:3000";