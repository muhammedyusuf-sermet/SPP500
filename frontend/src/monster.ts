export enum Size {
	Tiny = "Tiny",
	Small = "Small",
	Medium = "Medium",
	Large = "Large",
	Huge = "Huge",
	Gargantuan = "Gargantuan"
}

export enum MonsterType {
	Beast = "Beast",
	Dragon = "Dragon",
	Monstrosity = "Monstrosity",
	Humanoid = "Humanoid",
	Fiend = "Fiend",
	Undead = "Undead",
	Giant = "Giant",
	Elemental = "Elemental",
	SwarmOfTinyBeasts = "SwarmOfTinyBeasts",
	Construct = "Construct",
	Celestial = "Celestial",
	Aberration = "Aberration",
	Fey = "Fey",
	Plant = "Plant",
	Ooze = "Ooze"
}

export enum MonsterRace {
	AnyRace = "AnyRace",
	Devil = "Devil",
	Demon = "Demon",
	Human = "Human",
	Shapechanger = "Shapechanger",
	Goblinoid = "Goblinoid",
	Titan = "Titan",
	Gnoll = "Gnoll",
	Gnome = "Gnome",
	Dwarf = "Dwarf",
	Elf = "Elf",
	Orc = "Orc",
	Kobold = "Kobold",
	Lizardfolk = "Lizardfolk",
	Merfolk = "Merfolk",
	Sahuagin = "Sahuagin",
	Grimlock = "Grimlock"
}

export enum Alignment {
	Unaligned = "Unaligned",
	AnyAlignment = "AnyAlignment",
	AnyNonGoodAlignment = "AnyNonGoodAlignment",
	AnyNonEvilAlignment = "AnyNonEvilAlignment",
	AnyNonLawfulAlignment = "AnyNonLawfulAlignment",
	AnyNonChaoticAlignment = "AnyNonChaoticAlignment",
	AnyGoodAlignment = "AnyGoodAlignment",
	AnyEvilAlignment = "AnyEvilAlignment",
	AnyLawfulAlignment = "AnyLawfulAlignment",
	AnyChaoticAlignment = "AnyChaoticAlignment",
	AnyNeutralAlignment = "AnyNeutralAlignment",
	LawfulGood = "LawfulGood",
	LawfulNeutral = "LawfulNeutral",
	LawfulEvil = "LawfulEvil",
	NeutralGood = "NeutralGood",
	Neutral = "Neutral",
	NeutralEvil = "NeutralEvil",
	ChaoticGood = "ChaoticGood",
	ChaoticNeutral = "ChaoticNeutral",
	ChaoticEvil = "ChaoticEvil"
}

export enum Environment {
	Arctic = "Arctic",
	Coastal = "Coastal",
	Desert = "Desert",
	Forest = "Forest",
	Grassland = "Grassland",
	Hill = "Hill",
	Mountain = "Mountain",
	Swamp = "Swamp",
	Underdark = "Underdark",
	Underwater = "Underwater",
	Urban = "Urban"
}

export enum MonsterAction {
	SpecialAbility = "SpecialAbility",
	Action = "Action",
	LegendaryAction = "LegendaryAction"
}

export interface IMonsterAbilities {
	name: string,
	description: string,
}

export enum TypeOfAction {
	Ranged = "Ranged",
	Melee = "Melee",
}

export interface IMonsterActions {
	name: string,
	description: string,
	usageCount: number,
	legendaryAction: boolean,
	typeOfAction?: TypeOfAction,
	toHit: number,
	reach: number,
	targets: number,
	averagedDamage: number,
	damageDice: number,
	damageConstant: number,

}

export interface IMonsterState {
	Id?: number,
	Name: string,
	Size?: Size | string;
	Type?: MonsterType | string;
	Race?: MonsterRace | string;
	Environment?: Environment | string;
	Alignment?: Alignment | string;

	ArmorClass?: number;
	HitPoints?: number;
	HitPointDistribution?: string;

	Speed?: string;
	Languages?: string;

	DamageVulnerabilities?: string;
	DamageResistances?: string;
	DamageImmunities?: string;
	ConditionImmunities?: string;

	ChallengeRating?: number;

	AbilityScores: IMonsterAbilityScoreState;
	Senses: SenseMap;
	Skills: SkillMap;
	SavingThrows: IMonsterSavingThrowState;
	//Actions: IActionState[];
}

export interface SkillMap {
	[skillName: string]: number|undefined;
}

export interface SenseMap {
	[senseName: string]: number|undefined;
}

export interface IMonsterAbilityScoreState {
	[index:string]: number|undefined;
	Strength?: number;
	Dexterity?: number;
	Constitution?: number;
	Intelligence?: number;
	Wisdom?: number;
	Charisma?: number;
}

export interface IMonsterSavingThrowState {
	[index:string]: number|undefined;
	Strength?: number;
	Dexterity?: number;
	Constitution?: number;
	Intelligence?: number;
	Wisdom?: number;
	Charisma?: number;
}

export interface IActionState {
	Name: string,
	Description: string,
	HitBonus?: number,
	Damage?: string,
	DamageBonus?: number,
	Type?: MonsterAction
}
/*export interface IMonster {
	name: string,
	type: MonsterType,
	alignment: MonsterAlignment,
	size: MonsterSize,
	race: MonsterRace,
	environment: MonsterEnvironment,
	resistance?: string,
	damageImmunity?: string,
	conditionImmunity?: string,
	vulnerability?: string,
	armorClass: number,
	hitPoints?: number,
	hitPointDice?: string,
	hitPointDiceAdd?: number,
	speedLand: number,
	speedSwim?: number,
	strStat: number,
	dexStat: number,
	conStat: number,
	intStat: number,
	wisStat: number,
	chaStat: number,
	strSavingThrow?: number,
	dexSavingThrow?: number,
	conSavingThrow?: number,
	intSavingThrow?: number,
	wisSavingThrow?: number,
	chaSavingThrow?: number,
	skillsAthletics: number,
	skillsAcrobatics?: number,
	skillsSleightOfHand?: number,
	skillsStealth?: number,
	skillsArcana?: number,
	skillsHistory?: number,
	skillsInvestigation?: number,
	skillsNature?: number,
	skillsReligion?: number,
	skillsAnimalHandling?: number,
	skillsInsight?: number,
	skillsMedicine?: number,
	skillsPerception?: number,
	skillsSurvival?: number,
	skillsDeception?: number,
	skillsIntimidation?: number,
	skillsPerformance?: number,
	skillsPersuasion?: number,
	sensesBlindsight?: number,
	sensesDarkvision?: number,
	sensesTremorsense?: number,
	sensesTruesight?: number,
	sensesPassivePerception?: number,
	sensesPassiveInvestigation?: number,
	sensesPassiveInsight?: number,
	languages?: string,
	challengeRating: number,
	experiencePoints: number,
	abilities?: IMonsterAbilities[],
	actions?: IMonsterActions[],
}*/