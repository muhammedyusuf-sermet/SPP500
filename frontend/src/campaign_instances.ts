import * as Campaign from "./campaign";

// Dummy array of monsters
import EncounterInstances from './encounter_instances';

let encounterExample: Campaign.ICampaignState;
encounterExample = {
	Id: 0,
	Name: "Campaign",
	Summary: "Bloody hell...",
	Notes: "You gonna die!",
	Encounters: EncounterInstances['EncounterInstances'],
};

let CampaignInstances = new Array();

// Generate a dummy array of encounters with different IDs
for (var i = 0; i < 12; i++) {
	let encounter = {} as Campaign.ICampaignState;
	Object.assign(encounter, encounterExample);
	encounter['Id'] = i;
	CampaignInstances.push(encounter);
}

export default {
    CampaignInstances,
}