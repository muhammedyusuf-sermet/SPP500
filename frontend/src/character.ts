import { ICampaignData } from "./campaign";

export enum CharacterRace {
	Dragonborn = "Dragonborn",
	Dwarf = "Dwarf",
	Elf = "Elf",
	Gnome = "Gnome",
	HalfElf = "HalfElf",
	Halfling = "Halfling",
	HalfOrc = "HalfOrc",
	Human = "Human",
	Tiefling = "Tiefling"
}

export enum CharacterClass {
	Barbarian = "Barbarian",
	Bard = "Bard",
	Cleric = "Cleric",
	Druid = "Druid",
	Fighter = "Fighter",
	Monk = "Monk",
	Paladin = "Paladin",
	Ranger = "Ranger",
	Rogue = "Rogue",
	Sorcerer = "Sorcerer",
	Warlock = "Warlock",
	Wizard = "Wizard"
}

export interface ICharacterData {
	Id?: number;
	Name: string;
	Level?: number;
	Race?: CharacterRace;
	Class?: CharacterClass;
	MaxHealth?: number;
	ArmorClass?: number;
	Notes?: string;

	Campaigns?: ICampaignData[];
}
