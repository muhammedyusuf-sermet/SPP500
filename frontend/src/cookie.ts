import { API_URL } from "./config";
import Store from 'electron-store';

export class CookieManager {
	private constructor(){
	}

	private static store = new Store<string>();

	public static UserToken(cookieToCheck:string):string {
		// undefined if user is not logged in.
		var cookieVal = CookieManager.store.get(cookieToCheck);

		if (cookieVal === undefined)
			return cookieVal

		var request = require("request");
		var options = { method: 'GET',
			url: API_URL + '/verify',
			timeout: 2000,
			headers:
			{
				'Cache-Control': 'no-cache',
				'Content-Type': 'application/json',
				'Authorization': cookieVal
			},
			body: {},
			json: true
		};

		request(options, function (error: string, response: string, body: {status: number, message: string}) {
			if (error) {
				return undefined
			}
			else {
				return (body.status === 201) ? "Valid cookie" : undefined // Success
			}
		});
		return CookieManager.store.get(cookieToCheck);
	}

	public static SetStringCookie(name:string, data:string) {
		CookieManager.store.set(name, data);
	}

	public static RemoveCookie(name:string) {
		CookieManager.store.delete(name);
	}
}
