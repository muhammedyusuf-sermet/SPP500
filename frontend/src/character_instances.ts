import { CharacterRace, CharacterClass, ICharacterData } from "./character";


let characterExample: ICharacterData;
characterExample = {
	Id: 1,
	Name: "Character",
	Level: 3,
	Race: CharacterRace.Gnome,
	Class: CharacterClass.Paladin,
	MaxHealth: 23,
	ArmorClass: 12,
	Notes: 'aggressive',

	Campaigns: []
};

export const CharacterInstances: ICharacterData[] = new Array();

// Generate a dummy array of characters with different IDs
for (var i = 0; i < 12; i++) {
	let character = {} as ICharacterData;
	Object.assign(character, characterExample);
	character.Id = i + 1;
	CharacterInstances.push(character);
}