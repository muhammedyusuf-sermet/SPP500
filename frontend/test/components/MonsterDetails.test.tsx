import * as React from "react"
import { shallow, ShallowWrapper } from 'enzyme';
import {MonsterDetails, IMonsterDetailsState} from '../../src/renderer/components/platform/pages/view_catalog/MonsterDetails';
import * as Monster from '../../src/monster';

describe('Test the Monster View Details Per Monster', () => {
	let monsterDetailsInstance: ShallowWrapper<any, IMonsterDetailsState, MonsterDetails>;

	beforeEach(() => {
		// This mock object is in the same format with the backend would return
		// Backend currently doesn't return skills, senses, etc.
		let monsterExample = {
			Id: "",
			Name: "Hello",
			Type: Monster.MonsterType.Celestial,
			Alignment: Monster.Alignment.AnyGoodAlignment,
			Size: Monster.Size.Gargantuan,
			Race: Monster.MonsterRace.Devil,
			Environment: Monster.Environment.Underdark,
			DamageVulnerabilities: "Everything",
			DamageResistances: "None at all",
			DamageImmunities: "Nada",
			ConditionImmunities: "Nothing",
			ArmorClass: 15,
			HitPoints: 40,
			HitPointDistribution: "9d5 - 5",
			AbilityScores: null,
			SavingThrows: null,
			Skills: null,
			Senses: null,
			Languages: "Common and Draconic",
			ChallengeRating: 2.5,
		};
		monsterDetailsInstance = shallow(<MonsterDetails monster={monsterExample}/>);
	})

	it('renders without crashing', () => {
		expect(monsterDetailsInstance).toBeDefined();
	});

	it('renders correctly when the page is loaded', () => {
		expect(monsterDetailsInstance).toMatchSnapshot();
	});

	it('should return true if the IMonsterState object is empty', () => {
		let selectedMonster: Monster.IMonsterState;
		selectedMonster = {} as Monster.IMonsterState;
		let isEmpty = monsterDetailsInstance.instance().isEmptyObject(selectedMonster);

		expect(isEmpty).toEqual(true);
	});

});
