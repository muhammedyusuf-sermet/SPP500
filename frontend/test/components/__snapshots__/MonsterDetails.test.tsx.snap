// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Monster View Details Per Monster renders correctly when the page is loaded 1`] = `
<form>
  <h1
    className="page-title"
  >
    Monster Details
  </h1>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Monster Name
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      id="name"
      label="Monster Name"
      readOnly={true}
      value="Hello"
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Tile)
    className="box"
    isVertical={true}
  >
    <withHelpersModifiers(Subtitle)>
      Basic Configurations
    </withHelpersModifiers(Subtitle)>
    <withHelpersModifiers(Control)>
      <withHelpersModifiers(Label)>
        Type
      </withHelpersModifiers(Label)>
      <withHelpersModifiers(Input)
        label="Type"
        readOnly={true}
        value="Celestial"
      />
    </withHelpersModifiers(Control)>
    <withHelpersModifiers(Control)>
      <withHelpersModifiers(Label)>
        Size
      </withHelpersModifiers(Label)>
      <withHelpersModifiers(Input)
        label="Size"
        readOnly={true}
        value="Gargantuan"
      />
    </withHelpersModifiers(Control)>
    <withHelpersModifiers(Control)>
      <withHelpersModifiers(Label)>
        Race
      </withHelpersModifiers(Label)>
      <withHelpersModifiers(Input)
        label="Race"
        readOnly={true}
        value="Devil"
      />
    </withHelpersModifiers(Control)>
    <withHelpersModifiers(Control)>
      <withHelpersModifiers(Label)>
        Alignment
      </withHelpersModifiers(Label)>
      <withHelpersModifiers(Input)
        label="Alignment"
        readOnly={true}
        value="AnyGoodAlignment"
      />
    </withHelpersModifiers(Control)>
    <withHelpersModifiers(Control)>
      <withHelpersModifiers(Label)>
        Environment
      </withHelpersModifiers(Label)>
      <withHelpersModifiers(Input)
        label="Environment"
        readOnly={true}
        value="Underdark"
      />
    </withHelpersModifiers(Control)>
  </withHelpersModifiers(Tile)>
  <withHelpersModifiers(Tile)
    className="box"
    isVertical={true}
  >
    <withHelpersModifiers(Subtitle)>
      Vulnerability, Resistance, and Immunity
    </withHelpersModifiers(Subtitle)>
    <withHelpersModifiers(Control)>
      <withHelpersModifiers(Label)>
        Resistance
      </withHelpersModifiers(Label)>
      <withHelpersModifiers(Input)
        label="Resistance"
        readOnly={true}
        value="None at all"
      />
    </withHelpersModifiers(Control)>
    <withHelpersModifiers(Control)>
      <withHelpersModifiers(Label)>
        Damage Immunity
      </withHelpersModifiers(Label)>
      <withHelpersModifiers(Input)
        label="Damage Immunity"
        readOnly={true}
        value="Nada"
      />
    </withHelpersModifiers(Control)>
    <withHelpersModifiers(Control)>
      <withHelpersModifiers(Label)>
        Condition Immunity
      </withHelpersModifiers(Label)>
      <withHelpersModifiers(Input)
        label="Condition Immunity"
        readOnly={true}
        value="Nothing"
      />
    </withHelpersModifiers(Control)>
    <withHelpersModifiers(Control)>
      <withHelpersModifiers(Label)>
        Damage Vulnerabilities
      </withHelpersModifiers(Label)>
      <withHelpersModifiers(Input)
        label="Vulnerability"
        readOnly={true}
        value="Everything"
      />
    </withHelpersModifiers(Control)>
  </withHelpersModifiers(Tile)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Armor Class
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Armor Class"
      readOnly={true}
      value={15}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Hit Points
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Hit Points"
      readOnly={true}
      value={40}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Hit Point Distribution
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Hit Point Dice"
      readOnly={true}
      value="9d5 - 5"
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Strength Stat
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Strength Stat"
      readOnly={true}
      value={17}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Dexterity Stat
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Dexterity Stat"
      readOnly={true}
      value={15}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Constitution Stat
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Constitution Stat"
      readOnly={true}
      value={13}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Intelligence Stat
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Intelligence Stat"
      readOnly={true}
      value={12}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Wisdom Stat
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Wisdom Stat"
      readOnly={true}
      value={16}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Charisma Stat
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Charisma Stat"
      readOnly={true}
      value={15}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Strength Saving Throw Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Strength Saving Throw Modifier"
      readOnly={true}
      value={-3}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Dexterity Saving Throw Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Dexterity Saving Throw Modifier"
      readOnly={true}
      value={0}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Constitution Saving Throw Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Constitution Saving Throw Modifier"
      readOnly={true}
      value={-1}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Intelligence Saving Throw Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Intelligence Saving Throw Modifier"
      readOnly={true}
      value={-2}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Wisdom Saving Throw Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Wisdom Saving Throw Modifier"
      readOnly={true}
      value={8}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Charisma Saving Throw Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Charisma Saving Throw Modifier"
      readOnly={true}
      value={9}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Athletics Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Athletics Modifier"
      readOnly={true}
      value={9}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Acrobatics Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Acrobatics Modifier"
      readOnly={true}
      value={10}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Sleight of Hand Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Sleight of Hand Modifier"
      readOnly={true}
      value={9}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Stealth Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Stealth Modifier"
      readOnly={true}
      value={8}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Arcana Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Arcana Modifier"
      readOnly={true}
      value={7}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      History Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="History Modifier"
      readOnly={true}
      value={7}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Investigation Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Investigation Modifier"
      readOnly={true}
      value={6}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Nature Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Nature Modifier"
      readOnly={true}
      value={7}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Religion Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Religion Modifier"
      readOnly={true}
      value={8}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Animal Handling Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Animal Handling Modifier"
      readOnly={true}
      value={9}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Insight Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Insight Modifier"
      readOnly={true}
      value={10}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Medicine Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Medicine Modifier"
      readOnly={true}
      value={12}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Perception Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Perception Modifier"
      readOnly={true}
      value={15}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Survival Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Survival Modifier"
      readOnly={true}
      value={11}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Deception Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Deception Modifier"
      readOnly={true}
      value={10}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Intimidation Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Intimidation Modifier"
      readOnly={true}
      value={9}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Performance Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Performance Modifier"
      readOnly={true}
      value={7}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Persuasion Modifier
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Persuasion Modifier"
      readOnly={true}
      value={4}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Languages
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Languages"
      readOnly={true}
      value="Common and Draconic"
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Control)>
    <withHelpersModifiers(Label)>
      Challenge Rating
    </withHelpersModifiers(Label)>
    <withHelpersModifiers(Input)
      label="Challenge Rating"
      readOnly={true}
      value={2.5}
    />
  </withHelpersModifiers(Control)>
  <withHelpersModifiers(Button)
    className="button"
    disabled={true}
    type="submit"
  >
     Edit Monster 
  </withHelpersModifiers(Button)>
  <withHelpersModifiers(Button)
    className="button"
  >
     Return to Catalogue 
  </withHelpersModifiers(Button)>
</form>
`;
